sklearn/ensemble/bagging.py

```diff
@@ -65,7 +65,7 @@ def _parallel_build_estimators(n_estimators, ensemble, X, y, sample_weight,
             print("building estimator %d of %d" % (i + 1, n_estimators))
 
         random_state = check_random_state(seeds[i])
-        seed = check_random_state(random_state.randint(MAX_INT))
+        seed = random_state.randint(MAX_INT)
```

Comment:
As it is now `seed` is an integer, if we revert the change it will be a `random_state` object again. The `seeds` argument to `_parallel_build_estimators()` is indeed an array of `int`s, this is the argument that is being serialised by joblib parallel no?

Vague discussion summary: I asked at the sprint why my `sklearn.ensemble.tests.test_bagging.test_warm_start` was failing, we discussed a bit and couldn't find a good reason and as part of it decided that it seemed weird to have `seed` be a `random_state` derived from a `random_state` by seeding it, drawing a random int and using that as a seed. My memory is quite vague, so we should wait for @glouppe to say something.

Observation: reverting the change causes `sklearn.ensemble.tests.test_bagging.test_warm_start` to fail. Is L596 of `test_bagging.py` the right way to check that the trees have the same random state?